idea-plugin/src/test/kotlin/arrow/meta/ide/plugins/comprehensions/ComprehensionsTest.kt

```diff
@@ -1,75 +1,75 @@
-          it.takeIf { descriptor -> descriptor.lineMarker.size == 2 && descriptor.slowLM.isEmpty() }
-        }
-      ),
-      IdeTest(
-        code = ComprehensionsTestCode.code2,
-        myFixture = myFixture,
-        test = { code, myFixture ->
-          collectLM(code, myFixture, ArrowIcons.BIND)
-        },
-        result = resolves("LineMarkerTest for two LM on untyped variables") {
-          it.takeIf { descriptor -> descriptor.lineMarker.size == 2 && descriptor.slowLM.isEmpty() }
-        }
-      ),
-      IdeTest(
-        code = ComprehensionsTestCode.code3,
-        myFixture = myFixture,
-        test = { code, myFixture ->
-          collectLM(code, myFixture, ArrowIcons.BIND)
-        },
-        result = resolves("LineMarkerTest for six LM on untyped variables") {
-          it.takeIf { descriptor -> descriptor.lineMarker.size == 6 && descriptor.slowLM.isEmpty() }
-        }
-      ),
-      IdeTest(
-        code = ComprehensionsTestCode.code4,
-        myFixture = myFixture,
-        test = { code, myFixture ->
-          collectLM(code, myFixture, ArrowIcons.BIND)
-        },
-        result = resolves("LineMarkerTest for four LM on untyped variables") {
-          it.takeIf { descriptor -> descriptor.lineMarker.size == 4 && descriptor.slowLM.isEmpty() }
-        }
-      ),
-      IdeTest(
-        code = ComprehensionsTestCode.code5,
-        myFixture = myFixture,
-        test = { code, myFixture ->
-          collectLM(code, myFixture, ArrowIcons.BIND)
-        },
-        result = resolves("LineMarkerTest for zero LM ") {
-          it.takeIf { descriptor -> descriptor.lineMarker.isEmpty() && descriptor.slowLM.isEmpty() }
-        }
-      ),
-      IdeTest(
-        code = ComprehensionsTestCode.code6,
-        myFixture = myFixture,
-        test = { code, myFixture ->
-          collectLM(code, myFixture, ArrowIcons.BIND)
-        },
-        result = resolves("LineMarkerTest for no LM ") {
-          it.takeIf { descriptor -> descriptor.lineMarker.isEmpty() && descriptor.slowLM.isEmpty() }
-        }
+      myFixture = myFixture,
+      ctx = IdeMetaPlugin()
+    ) {
+      listOf<IdeTest<LineMarkerDescription, IdeMetaPlugin>>(
+        IdeTest(
+          code = ComprehensionsTestCode.code1,
+          test = { code, myFixture, _ ->
+            collectLM(code, myFixture, ArrowIcons.BIND)
+          },
+          result = resolvesWith("LineMarkerTest for two LM on typed variables") {
+            it.takeIf { descriptor -> descriptor.lineMarker.size == 2 && descriptor.slowLM.isEmpty() }
+          }
+        ),
+        IdeTest(
+          code = ComprehensionsTestCode.code2,
+          test = { code, myFixture, _ ->
+            collectLM(code, myFixture, ArrowIcons.BIND)
+          },
+          result = resolvesWith("LineMarkerTest for two LM on untyped variables") {
+            it.takeIf { descriptor -> descriptor.lineMarker.size == 2 && descriptor.slowLM.isEmpty() }
```

Comment:
can these repeated expressions be refactored to a function?, they all seem to do the same except change the source code.

Reply:
Yeah for sure, I'll add that to the new PR